BlobTierAnalysisTool/BlobTierAnalysisTool.NetCore/Helpers/BlobStorageHelper.cs

```diff
@@ -242,22 +201,23 @@ public static string StorageAccountName
         /// <param name="blob"></param>
         /// <param name="filterCriteria"></param>
         /// <returns></returns>
-        private static bool DoesBlobMatchFilterCriteria(CloudBlockBlob blob, Models.FilterCriteria filterCriteria)
+        private static bool DoesBlobMatchFilterCriteria(BlobClient blob, Models.FilterCriteria filterCriteria)
         {
-            if (blob.Properties.StandardBlobTier == StandardBlobTier.Archive) return false;
+            BlobProperties blobProperties = blob.GetProperties().Value;
```

Comment:
Any reason to not use async here? Also there is implicit cast from Response<T> to T, so the .Value should be unnecessary.